Test for mc_invalid_rate fails

**Describe the bug**
The test for multiprocessing with invalid order of rate fails sometimes (i.e succeeds in running simulation). 

**To Reproduce**
Steps to reproduce the behavior:
1. Run tests on backend
2. Some of the times it fails

**Expected behavior**
The test should always give an invalid order of rate and thus cause the simulation to fail. 


**Desktop (please complete the following information):**
 - OS: Linux

**Additional context**
Captured log:
_____________________ test_run_mc_sim_rate_invalid_order ______________________

client = <starlette.testclient.TestClient object at 0x7f81694[34](https://github.com/obennet/sed-backend/actions/runs/4501057694/jobs/7920980127#step:6:35)a60>
std_headers = {'Authorization': '***'}
std_user = UserPost(username='test_user_9nz8idguw', email='test_user_9nz8idguw@sed-mock-email.com', full_name=None, disabled=False, password='7r44lnsw472mzl2vjup8', scopes='user')

    def test_run_mc_sim_rate_invalid_order(client, std_headers, std_user):
        #Setup
      current_user = impl_users.impl_get_user_with_username(std_user.username)
    
      project, vcs, design_group, design, settings = sim_tu.setup_single_simulation(current_user.id)
      tu.edit_rate_order_formulas(project.id, vcs.id, design_group.id)
      settings.monte_carlo = False
    
      #Act
      res = client.post(f'/api/cvs/project/{project.id}/simulation/run-multiprocessing',
                        headers=std_headers,
                        json = {
                          "sim_settings": settings.dict(),
                          "vcs_ids": [vcs.id],
                          "design_ids": [design[0].id]
                        })
    
      #Assert
>     assert res.status_code == 400
E     assert 200 == 400
E      +  where 200 = <Response [200]>.status_code

tests/apps/cvs/simulation/test_sim_multiprocessing.py:233: AssertionError
----------------------------- Captured stdout call -----------------------------
Seed settings vcs rows ['Transition', 'Information management', 'Quality management', 'Risk management', 'mo7um2eqsov4ygf1bzn', 'Human resource management', 'Life-cycle model management', 'Project assessment and control', 'Infrastructure management', 'Project portfolio management']
New rows ['Transition', 'Information management', 'Quality management', 'Risk management', 'mo7um2eqsov4ygf1bzn', 'Human resource management', 'Life-cycle model management', 'Project assessment and control', 'Infrastructure management', 'Project portfolio management']
Common elements ['Transition', 'Information management', 'Quality management', 'Risk management', 'mo7um2eqsov4ygf1bzn', 'Human resource management', 'Life-cycle model management', 'Project assessment and control', 'Infrastructure management', 'Project portfolio management']
Row len 10

--- a/tests/apps/cvs/simulation/test_sim_multiprocessing.py
+++ b/tests/apps/cvs/simulation/test_sim_multiprocessing.py
@@ -212,13 +212,15 @@ def test_run_mc_sim_both_flows(client, std_headers, std_user):
   tu.delete_vd_from_user(current_user.id)
 
 
-'''
+
 def test_run_mc_sim_rate_invalid_order(client, std_headers, std_user):
     #Setup 
   current_user = impl_users.impl_get_user_with_username(std_user.username)
 
   project, vcs, design_group, design, settings = sim_tu.setup_single_simulation(current_user.id)
-  tu.edit_rate_order_formulas(project.id, vcs.id, design_group.id)
+  first_tech_process = tu.edit_rate_order_formulas(project.id, vcs.id, design_group.id)
+  if first_tech_process is None:
+    raise Exception("Cannot find first technical process")
   settings.monte_carlo = False
 
   #Act
@@ -240,4 +242,3 @@ def test_run_mc_sim_rate_invalid_order(client, std_headers, std_user):
   tu.delete_VCS_with_ids(project.id, [vcs.id])
   tu.delete_project_by_id(project.id, current_user.id)
   tu.delete_vd_from_user(current_user.id)
-'''

--- a/tests/apps/cvs/simulation/testutils.py
+++ b/tests/apps/cvs/simulation/testutils.py
@@ -11,7 +11,7 @@ def setup_single_simulation(user_id) -> Tuple[CVSProject, VCS, DesignGroup, List
   project = tu.seed_random_project(user_id)
   vcs = tu.seed_random_vcs(project.id)
   design_group = tu.seed_random_design_group(project.id)
-  tu.seed_random_formulas(project.id, vcs.id, design_group.id, user_id, 10) #Also creates the vcs rows
+  tu.seed_random_formulas(project.id, vcs.id, design_group.id, user_id, 15) #Also creates the vcs rows
   design = tu.seed_random_designs(project.id, design_group.id, 1)
 
   settings = tu.seed_simulation_settings(project.id, [vcs.id], [design[0].id])

--- a/tests/apps/cvs/testutils.py
+++ b/tests/apps/cvs/testutils.py
@@ -153,7 +153,10 @@ def random_table_row(
         subprocess = random_subprocess(project_id, vcs_id)
         subprocess_id = subprocess.id
     else:
-        iso_process_id = random.randint(1, 25)
+        if random.randint(1, 5) == 1: #Give 1/5 chance to produce non-tech process
+            iso_process_id = random.randint(1, 14)
+        else:
+            iso_process_id = random.randint(15, 25)
 
     if stakeholder is None:
         stakeholder = tu.random_str(5, 50)
@@ -556,10 +559,10 @@ def edit_rate_order_formulas(project_id: int, vcs_id: int, design_group_id: int)
     rows.reverse()  # reverse back to find first technical process
     for row in rows:
         if row.iso_process is not None:
-            if row.iso_process.category == 'Technical processes':
+            if row.iso_process.category == 'Technical processes' and row.id != last_id:
                 return row
         else:
-            if row.subprocess.parent_process.category == 'Technical processes':
+            if row.subprocess.parent_process.category == 'Technical processes' and row.id != last_id:
                 return row
 
 

